com.unity.render-pipelines.universal/Runtime/2D/Shadows/ShadowCasterGroup2DManager.cs

```diff
@@ -2,15 +2,36 @@
 using System.Collections.Generic;
 using UnityEngine;
 
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
 namespace UnityEngine.Experimental.Rendering.Universal
 {
+#if UNITY_EDITOR
+    [InitializeOnLoadAttribute]
+#endif
     internal class ShadowCasterGroup2DManager
     {
         static List<ShadowCasterGroup2D> s_ShadowCasterGroups = null;
 
         public static List<ShadowCasterGroup2D> shadowCasterGroups { get { return s_ShadowCasterGroups; } }
 
 
+#if UNITY_EDITOR
+        static ShadowCasterGroup2DManager()
+        {
+            EditorApplication.playModeStateChanged += OnPlayModeStateChanged;
+        }
+
+        private static void OnPlayModeStateChanged(PlayModeStateChange state)
+        {
+            if (state == PlayModeStateChange.ExitingEditMode || state == PlayModeStateChange.ExitingPlayMode)
```

Comment:
You should add a null check for s_ShadowCasterGroups here as it was causing a NullReferenceException